Terminal.Gui/Windows/FileDialog2.cs

```diff
@@ -0,0 +1,2118 @@
+
+			};
+
+			this.AllowsMultipleSelection = false;
+
+			this.UpdateNavigationVisibility ();
+
+			// Determines tab order
+			this.Add (this.btnToggleSplitterCollapse);
+			this.Add (this.tbFind);
+			this.Add (this.spinnerLabel);
+			this.Add (lblFeedback);
+			this.Add (this.btnOk);
+			this.Add (this.btnCancel);
+			this.Add (this.lblUp);
+			this.Add (this.lblBack);
+			this.Add (this.lblForward);
+			this.Add (lblPath);
+			this.Add (this.tbPath);
+			this.Add (this.splitContainer);
+		}
+		/// <inheritdoc/>
+		public override bool ProcessHotKey (KeyEvent keyEvent)
+		{
+			if (this.NavigateIf (keyEvent, Key.CtrlMask | Key.F, this.tbFind)) {
+				return true;
+			}
+
+			return base.ProcessHotKey (keyEvent);
+		}
+		private void RestartSearch ()
+		{
+			if (disposed || state?.Directory == null) {
+				return;
+			}
+
+			if (state is SearchState oldSearch) {
+				oldSearch.Cancel ();
+			}
+
+			// user is clearing search terms
+			if (tbFind.Text == null || tbFind.Text.Length == 0) {
+
+				// Wait for search cancellation (if any) to finish
+				// then push the current dir state
+				lock (onlyOneSearchLock) {
+					PushState (new FileDialogState (state.Directory, this), false);
+				}
+				return;
+			}
+
+			PushState (new SearchState (state?.Directory, this, tbFind.Text.ToString ()), true);
+		}
+
+		/// <inheritdoc/>
+		protected override void Dispose (bool disposing)
+		{
+			disposed = true;
+			base.Dispose (disposing);
+
+			if (state is SearchState search) {
+				search.Cancel ();
+			}
+		}
+
+		private void ClearFeedback ()
+		{
+			lblFeedback.Text = string.Empty;
+		}
+
+		private void CycleToNextTableEntryBeginningWith (KeyEventEventArgs keyEvent)
+		{
+			if (tableView.Table.Rows.Count == 0) {
+				return;
+			}
+
+			var row = tableView.SelectedRow;
+
+			// There is a multi select going on and not just for the current row
+			if (tableView.GetAllSelectedCells ().Any (c => c.Y != row)) {
+				return;
+			}
+
+			int match = collectionNavigator.GetNextMatchingItem (row, (char)keyEvent.KeyEvent.KeyValue);
+
+			if (match != -1) {
+				tableView.SelectedRow = match;
+				tableView.EnsureValidSelection ();
+				tableView.EnsureSelectedCellIsVisible ();
+				keyEvent.Handled = true;
+			}
+		}
+
+		private void UpdateCollectionNavigator ()
+		{
+
+			var collection = tableView
+				.Table
+				.Rows
+				.Cast<DataRow> ()
+				.Select ((o, idx) => RowToStats (idx))
+				.Select (s => s.FileSystemInfo.Name)
+				.ToArray ();
+
+			collectionNavigator = new CollectionNavigator (collection);
+		}
+
+		/// <summary>
+		/// Gets or sets a value indicating whether to use Utc dates for date modified.
+		/// Defaults to <see langword="false"/>.
+		/// </summary>
+		public static bool UseUtcDates { get; set; } = false;
+
+		/// <summary>
+		/// Sets a <see cref="ColorScheme"/> to use for directories rows of
+		/// the <see cref="TableView"/>.
+		/// </summary>
+		public ColorScheme ColorSchemeDirectory { private get; set; }
```

Comment:
Shoud these all be configurable? For those that are default: In my early ConfigManager stuff, the existing examples of settings that are defaults, I went with `Default<propname>` for naming. Should we do this for these?